src/components/ExpenseDetails.tsx

```diff
@@ -83,11 +45,63 @@ export const ExpenseDetails = () => {
           }
       }
 
-    useEffect(()=>{
-        fetchDataByRow(userId).then((data)=>{
-            setExpenses(expensesGetYear(data));
-        });
-    },[]);
+    const expensesFilter = () => {
+        return expenses.filter((exp:ExpensesData)=>{
+            return (createPureMonthAndYearDataFormat(exp)).includes(idFormat);
+        }).map((expens:ExpensesData,i: number)=>{                    
+            return (
+                <>
+                        <div>{i+1}</div>
+                        <div>{expens.productCategory}</div>
+                        <div>{expens.productPrice} zł</div>
+                        <div>
+                            <Link to={`/expenseDetails/${expens.productLabel}/edit`}>
+                                <ConfirmButton value={buttonData.editButton} />
```

Comment:
A jaki miałaś problem z przekazywaniem tego parametru? :) 

Mi się wydaję że miałąbyś po pierwsze dodać nowy props do ConfirmButton (i w sumie do wszystkich przycisków chyba by się przydało :) ), potem tutaj przekazujesz reference do funkcji którą chcesz odpalić, a w ConfirmButton z kolei już definiujesz onClick=przekazanaReferencjaFunkcji

Omówmy to na spotkaniu :)